### PR TITLE
[Elastic Agent] Fix install with bootstrapping Fleet Server

### LINKED ISSUE #24950: [Elastic Agent] Install command does not work with Fleet Server bootstrap
Running the `install` command will prompt for questions it should not ask when running with `--fleet-server` parameter.

```
sudo ./elastic-agent install --fleet-server https://elastic:${pass}@fleet-server-enroll.es.us-central1.gcp.foundit.no:9243 --enrollment-token ${token}
Password:
The Elastic Agent is currently in BETA and should not be used in production

Elastic Agent will be installed at /Library/Elastic/Agent and will run as a service. Do you want to continue? [Y/n]:Y
Do you want to enroll this Agent into Fleet? [Y/n]:Y
URL you want to enroll this Agent into: ^C
```

If you force the install with `-f` it still does not perform the bootstrap of the Fleet Server.

```
sudo ./elastic-agent install --fleet-server https://elastic:${pass}@fleet-server-enroll.es.us-central1.gcp.foundit.no:9243 --enrollment-token ${token} -f
The Elastic Agent is currently in BETA and should not be used in production
```

This will just cause the Elastic Agent to be installed in standalone mode and not actually enroll or start Fleet Server.

To workaround for this is to perform `enroll` after the `install` command. Then the Elastic Agent will run Fleet Server and enroll into Fleet.

```
% cd /Library/Elastic/Agen
% sudo ./elastic-agent enroll --fleet-server https://elastic:${pass}@fleet-server-enroll.es.us-central1.gcp.foundit.no:9243 --enrollment-token ${token} -f
The Elastic Agent is currently in BETA and should not be used in production

2021-04-06T11:27:55.615-0400	INFO	cmd/enroll_cmd.go:281	Generating self-signed certificate for Fleet Server
2021-04-06T11:27:57.056-0400	INFO	cmd/enroll_cmd.go:527	Waiting for Elastic Agent to start Fleet Server
2021-04-06T11:27:58.063-0400	INFO	cmd/enroll_cmd.go:558	Fleet Server - Starting
2021-04-06T11:28:01.078-0400	INFO	cmd/enroll_cmd.go:541	Fleet Server - Running on default policy with Fleet Server integration
2021-04-06T11:28:02.472-0400	INFO	cmd/enroll_cmd.go:190	Successfully triggered restart on running Elastic Agent.
Successfully enrolled the Elastic Agent.
```

### LINKED ISSUE #24960: [Standalone Agent]:No "Agent Installed" confirmation message on successful standalone agent installation.
**Kibana version:** 7.13.0 Snapshot Kibana cloud environment

**Host OS and Browser version:** Windows, All

**Build Details:**
```
  Artifact link used: https://snapshots.elastic.co/7.13.0-bf9ad0ac/downloads/beats/elastic-agent/elastic-agent-7.13.0-SNAPSHOT-windows-x86_64.zip
  Build: 40068
  Commit: a455d0fa208238638ac744ff255a590ad3fb042a
```

**Preconditions:**
1. 7.13.0 Snapshot Kibana cloud environment should be available.

**Steps to reproduce:**
1. Login to Kibana environment.
2. Navigate to Agents tab>Add Agent>Run Standalone.
3. Follow the steps provided under "Run Standalone" section.
4. Run `./elastic-agent install` and observe no "Agent Installed" confirmation message.

**Expected Result:**
Agent installed confirmation message should be displayed on successful standalone agent installation.

**Screenshot:**

**Note:**
No impact on working of standalone agent is observed due to this issue.

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

This fixes the `install` command to allow Fleet Server to be bootstrapped properly. This also adds better logging to the `enroll` command (that is called by `install`) so its clear what might be causing the bootstrap of Fleet Server to fail.

This branch also uses a new feature from Fleet Server that removes the requirement of needing to pass the `--enrollment-token` to either `install` or `enroll`. Now Fleet Server will tell the Elastic Agent which enrollment token to use and Elastic Agent will use it to perform enrollment. This makes the installation of Elastic Agent with Fleet Server to just the following:

```
$ sudo ./elastic-agent install --fleet-server http://elastic:changeme@localhost:9200
```

This includes removing the requirement for the Docker container to even have to talk to Kibana when bootstrapping with Fleet Server. The following command would run the container and it would not even talk to Kibana at any point.

```
$ docker run -it -e ELASTICSEARCH_HOST=http://host.docker.internal:9200 -e FLEET_SERVER_ENABLE=1 docker.elastic.co/beats/elastic-agent:8.0.0-SNAPSHOT
```

This includes some fixes to the paths that was prevent Fleet Server from starting.

## Why is it important?

So bootstrapping of Fleet Server works with the `install` command and to simplify the bootstrapping process by removing the requirement for `--enrollment-token`.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes #24950
- Closes #24960

## Logs

```
% sudo ./elastic-agent install --fleet-server http://elastic:changeme@localhost:9200
The Elastic Agent is currently in BETA and should not be used in production

Elastic Agent will be installed at /Library/Elastic/Agent and will run as a service. Do you want to continue? [Y/n]:Y
2021-04-07T19:48:27.703-0400    INFO    cmd/enroll_cmd.go:289   Generating self-signed certificate for Fleet Server
2021-04-07T19:48:29.005-0400    INFO    cmd/enroll_cmd.go:579   Waiting for Elastic Agent to start Fleet Server
2021-04-07T19:48:30.007-0400    INFO    cmd/enroll_cmd.go:619   Fleet Server - Starting
2021-04-07T19:48:33.020-0400    INFO    cmd/enroll_cmd.go:593   Fleet Server - Running on default policy with Fleet Server integration; missing config fleet.agent.id
2021-04-07T19:48:34.657-0400    INFO    cmd/enroll_cmd.go:193   Successfully triggered restart on running Elastic Agent.
Successfully enrolled the Elastic Agent.
```

### CODE REVIEW

**Comment by blakerouse** [comment]
@ruflin @EricDavisX This removes the requirement of needing the `--enrollment-token` when bootstrapping with Fleet Server. All that is needed is:

```
sudo ./elastic-agent install --fleet-server http://elastic:changeme@localhost:9200
```

I am not marking this as breaking, because you can still provide one if you want. Its just no longer required to provide one.

File: x-pack/elastic-agent/pkg/agent/application/pipeline/emitter/modifiers/fleet_decorator.go [hunk 1]
```diff
@@ -32,19 +30,21 @@ func InjectFleet(cfg *config.Config, hostInfo types.HostInfo, agentInfo *info.Ag
...
-		if !ok {
-			return fmt.Errorf("failed to get api key from fleet config")
+		if ok {
+			nodes = append(nodes, token)
 		}
 
 		kbn, ok := transpiler.Lookup(ast, "fleet.kibana")
-		if !ok {
-			return fmt.Errorf("failed to get kibana config key from fleet config")
+		if ok {
```
Comment (michalpristas) [inline x-pack/elastic-agent/pkg/agent/application/pipeline/emitter/modifiers/fleet_decorator.go:41]: we can do inline now to reduce scope

File: x-pack/elastic-agent/pkg/agent/cmd/install.go [hunk 2]
```diff
@@ -114,13 +118,13 @@ func installCmd(streams *cli.IOStreams, cmd *cobra.Command, args []string) error
...
 	}
-	if !askEnroll && (url == "" || token == "") {
+	if !askEnroll && (url == "" || token == "") && fleetServer == "" {
 		// force was performed without required enrollment arguments, all done (standalone mode)
 		enroll = false
 	}
 
 	if enroll {
-		if url == "" {
+		if url == "" && fleetServer == "" {
```
Comment (michalpristas) [inline x-pack/elastic-agent/pkg/agent/cmd/install.go:127]: could we just move this to `enroll && fleetServer == ""`  and remove also from the if above
or if the goal here is to skip enroll for fleet server maybe we can just extract this up to the line 140 into a Fn and execute only when `fleetServer == ""` 
these ifs are getting complicated

File: x-pack/elastic-agent/pkg/agent/application/pipeline/emitter/modifiers/fleet_decorator.go (hunk 1 above)
Comment (blakerouse) [reply x-pack/elastic-agent/pkg/agent/application/pipeline/emitter/modifiers/fleet_decorator.go]: fixed

File: x-pack/elastic-agent/pkg/agent/cmd/install.go (hunk 2 above)
Comment (blakerouse) [reply x-pack/elastic-agent/pkg/agent/cmd/install.go]: I was able to simplify this by moving it up. The goal here is to ask for URL and Token in the case that it is not provided. The complexity comes from `install` being interactive.

**Comment by blakerouse** [comment]
@michalpristas Ready for another look

**Review by michalpristas — APPROVED** [review]
seems ok, tested with latest kibana

**Pull request merged**